lib/Dialect/RTL/RTLTypes.cpp

```diff
@@ -109,6 +121,29 @@ bool circt::rtl::hasRTLInOutType(Type type) {
   return type.isa<InOutType>();
 }
 
+/// Return the canonical type, that is, the type with all type aliases
+/// recursively resolved to a base type.
+Type circt::rtl::getCanonicalType(Type type) {
```

Comment:
This is recomputing the canonical type every call, and it's called often.  My recollection is all our types are going to need a canonical type pointer so we don't have to construct new types all the time.

Reply:
FIRRTL has a base type with a lot of functionality.  I haven't looked in to how to do that with ODS generated types.

Reply:
Good point. I believe the right thing to do to optimize this would be to store the canonical type in the TypeAlias type's storage upon construction. I can take a look into that.